### PR TITLE
feat: add ability to configure timeout in AWSClientConfiguration

### PR DESCRIPTION
**Pull request by dayaffe**
Add ability to configure connectTimeoutMs in AWSClientConfiguration.

## Issue \#
#1065 
Companion PR: [#592](https://github.com/smithy-lang/smithy-swift/pull/592)

## Description of changes
- expose connectTimeoutMs in AWSClientConfiguration which is used to create a CRTClientEngine() with the related configuration.

Ex. Usage (S3)
```
let client = try await S3Client(
    config: S3Client.S3ClientConfiguration(connectTimeoutMs: 15_000)
)
```

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Sources/Core/AWSClientRuntime/AWSClientConfiguration.swift [hunk 1]
```diff
@@ -102,6 +102,11 @@ public class AWSClientConfiguration<ServiceSpecificConfiguration: AWSServiceSpec
     /// This structure is custom code-generated for each AWS service.
     public var serviceSpecific: ServiceSpecificConfiguration
 
+    /// The timeout for a request in milliseconds
+    ///
+    /// If none is provided the client with use default values based on the operation system.
```
Comment (sichanyoo) [inline Sources/Core/AWSClientRuntime/AWSClientConfiguration.swift:107]: nit: typo on with?

**Review by sichanyoo — APPROVED** [review]
Lgtm, except one typo nit in comment

File: Sources/Core/AWSClientRuntime/AWSClientConfiguration.swift [hunk 2]
```diff
@@ -102,6 +102,11 @@ public class AWSClientConfiguration<ServiceSpecificConfiguration: AWSServiceSpec
     /// This structure is custom code-generated for each AWS service.
     public var serviceSpecific: ServiceSpecificConfiguration
 
+    /// The timeout for a request in milliseconds
+    ///
+    /// If none is provided the client will use default values based on the operation system.
```
Comment (jbelkins) [inline Sources/Core/AWSClientRuntime/AWSClientConfiguration.swift:107]: nit: based on the _platform_

File: Tests/Core/AWSClientRuntimeTests/AWSClientConfigurationTests.swift [hunk 3]
```diff
@@ -63,6 +63,14 @@ class AWSClientConfigurationTests: XCTestCase {
         let subject = try await Subject(region: region, appID: appID)
         XCTAssertEqual(subject.appID, appID)
     }
+
+    // MARK: - Timeout
+
+    func test_async_configureTimeoutOptionsFromParams() throws {
```
Comment (jbelkins) [inline Tests/Core/AWSClientRuntimeTests/AWSClientConfigurationTests.swift:69]: Do we also need a test for default timeout?

**Review by jbelkins — CHANGES_REQUESTED** [review]
A couple nits

File: Tests/Core/AWSClientRuntimeTests/AWSClientConfigurationTests.swift (hunk 3 above)
Comment (dayaffe) [reply Tests/Core/AWSClientRuntimeTests/AWSClientConfigurationTests.swift]: Adding one now. Will check for 30_000 or 3_000 depending on the operating system the test is run on. There doesn't seem to be a way to force the OS in XCTest

Comment (dayaffe) [reply Tests/Core/AWSClientRuntimeTests/AWSClientConfigurationTests.swift]: Removed this test as default timeout info is in CRT and not available to AWSClientConfiguration

**Review by jbelkins — APPROVED** [review]

**Pull request merged**